From a5747895ae28375a917e3c5f232abfce9b9cccc6 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Thu, 8 Nov 2012 18:35:09 +0100
Subject: [PATCH 3/3] firmware: Install firmware signature files automatically

Git-commit: Not yet, reviewing
Patch-mainline: Not yet, reviewing
References: fate#314574
Target: SLE-12

... when CONFIG_FIRMWARE_SIG is set.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
[2013-01-28: Add dependency on CONFIG_MODULE_SIG_ALL --mmarek]
---
 Makefile                |    7 +++++++
 scripts/Makefile.fwinst |   19 +++++++++++++++++--
 2 files changed, 24 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index b927add..e41f949 100644
--- a/Makefile
+++ b/Makefile
@@ -757,6 +757,13 @@ mod_sign_cmd = true
 endif
 export mod_sign_cmd
 
+fw_sign_cmd = true
+ifeq ($(CONFIG_FIRMWARE_SIG),y)
+ifdef CONFIG_MODULE_SIG_ALL
+fw_sign_cmd = perl $(srctree)/scripts/sign-file -f $(CONFIG_MODULE_SIG_HASH) $(MODSECKEY) $(MODPUBKEY)
+endif
+endif
+export fw_sign_cmd
 
 ifeq ($(KBUILD_EXTMOD),)
 core-y		+= kernel/ mm/ fs/ ipc/ security/ crypto/ block/
diff --git a/scripts/Makefile.fwinst b/scripts/Makefile.fwinst
index 4d908d1..86ad7cb 100644
--- a/scripts/Makefile.fwinst
+++ b/scripts/Makefile.fwinst
@@ -29,6 +29,21 @@ installed-mod-fw := $(addprefix $(INSTALL_FW_PATH)/,$(mod-fw))
 installed-fw := $(addprefix $(INSTALL_FW_PATH)/,$(fw-shipped-all))
 installed-fw-dirs := $(sort $(dir $(installed-fw))) $(INSTALL_FW_PATH)/./
 
+installed-fw-sig :=
+installed-mod-fw-sig :=
+ifdef CONFIG_MODULE_SIG_ALL
+ifdef CONFIG_FIRMWARE_SIG
+installed-fw-sig := $(patsubst %,%.sig, $(installed-fw))
+installed-mod-fw-sig := $(patsubst %,%.sig, $(installed-mod-fw))
+endif
+endif
+
+quiet_cmd_fwsig = FWSIG $@
+      cmd_fwsig = $(fw_sign_cmd) $(patsubst %.sig,%,$@) $@
+
+%.sig: %
+	$(call cmd,fwsig)
+
 # Workaround for make < 3.81, where .SECONDEXPANSION doesn't work.
 PHONY += $(INSTALL_FW_PATH)/$$(%) install-all-dirs
 $(INSTALL_FW_PATH)/$$(%): install-all-dirs
@@ -49,9 +64,9 @@ PHONY +=  __fw_install __fw_modinst FORCE
 
 .PHONY: $(PHONY)
 
-__fw_install: $(installed-fw)
+__fw_install: $(installed-fw) $(installed-fw-sig)
 
-__fw_modinst: $(installed-mod-fw)
+__fw_modinst: $(installed-mod-fw) $(installed-mod-fw-sig)
 	@:
 
 __fw_modbuild: $(addprefix $(obj)/,$(mod-fw))
-- 
1.6.4.2

